utils/ecl-diff/ecl-diff-component.js

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env node
```

Comment:
Putting this after including the suggestions below: you need to refactor this file to have a consistent coding style in terms of handling information flow. You have the following options:
1) Refactor all methods to promises and use `async/await`: straight-forward way of making changes.
2) Convert all methods to continuation passing style having callbacks in all cases. Not great, callback hell.
3) Use streams and chain operations piping results to std.out. I'd personally not do that as streams can be hard to debug and work with.